include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -49,16 +44,30 @@
 #    define _ONEDPL_PREDEFINED_POLICIES 1
 #endif
 
-#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND)
-#    define _ONEDPL_PAR_BACKEND_TBB 1
+// Check availability of parallel backends
+#if __has_include(<tbb/tbb.h>)
+#    define _ONEDPL_TBB_AVAILABLE 1
+#endif
+#if ONEDPL_USE_TBB_BACKEND && !_ONEDPL_TBB_AVAILABLE
+#    error "Parallel execution policies with oneTBB or Intel(R) TBB support are enabled, but the library is not found"
 #endif
 
-#if ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && defined(_OPENMP))
-#    define _ONEDPL_PAR_BACKEND_OPENMP 1
+#if defined(_OPENMP) && __has_include(<omp.h>)
+#    define _ONEDPL_OPENMP_AVAILABLE 1
+#endif
+// Avoid throwing an error during compilation for a device by checking __SYCL_DEVICE_ONLY__
+#if ONEDPL_USE_OPENMP_BACKEND && !_ONEDPL_OPENMP_AVAILABLE && !defined(__SYCL_DEVICE_ONLY__)
+#    error "Parallel execution policies with OpenMP* support are enabled, \
+        but OpenMP* headers are not found or the compiler does not support OpenMP*"
```

Comment:
Are we sure that in error messages we also need to put the asterisk after 3rd party trademarks?
In any case, it should be enough to do once, not three times.